adi/rx_tx.py

```diff
@@ -255,84 +252,62 @@ def __rx_unbuffered_data(self):
 
         return x
 
-    def __rx_buffered_data(self):
+    def __rx_buffered_data(self) -> Union[List[np.ndarray], np.ndarray]:
+        """__rx_buffered_data: Read data from RX buffer
+
+        Returns:
+            List of numpy arrays containing the data from the RX buffer that are
+            channel interleaved
+        """
         if not self.__rxbuf:
             self._rx_init_channels()
         self.__rxbuf.refill()
-        data = []
 
-        iio_enabled_channels = []
-        if self._complex_data is True:
+        data_channel_interleaved = []
+        ecn = []
+        if self._complex_data:
             for m in self.rx_enabled_channels:
-                iio_enabled_channels.append(self._rxadc.find_channel(self._rx_channel_names[m * 2]))
-                iio_enabled_channels.append(self._rxadc.find_channel(self._rx_channel_names[(m * 2) + 1]))
+                ecn.extend(
+                    (self._rx_channel_names[m * 2], self._rx_channel_names[m * 2 + 1])
+                )
         else:
-            for m in self.rx_enabled_channels:
-                iio_enabled_channels.append(self._rxadc.find_channel(self._rx_channel_names[m]))
-
-        iio_enabled_channels.sort(key=lambda ch: ch.index) # sort channels by index
-
-        data = self.__rxbuf.read() # read buffer from iio
-        # initialize and build dtype tuple for np.frombuffer
-        npdtype = ()
-        for ec in iio_enabled_channels:
-            df = ec.data_format
-            fmt = ">" if df.is_be is True else "<"
-            fmt += "i" if df.is_signed is True else "u"
-            fmt += str(df.length // 8)  # create format string -
-            npdtype = (*npdtype, fmt)
-        x = np.frombuffer(data, dtype=npdtype)
-
-        stride = len(iio_enabled_channels)
-        for idx, ec in enumerate(iio_enabled_channels):
-            df = ec.data_format
-            x[idx::stride] = np.right_shift(x[idx::stride], df.shift)
-            x[idx::stride] = np.bitwise_and(x[idx::stride], (1 << df.length) - 1)
-
-        if not self._rx_stack_interleaved and stride>1:
-            stacked_data = []
-            for idx, ec in enumerate(iio_enabled_channels):
-                stacked_data.extend(x[idx::stride])
-            x = stacked_data
-        return x
+            ecn = [self._rx_channel_names[m] for m in self.rx_enabled_channels]
+
+        for name in ecn:
+            chan = self._rxadc.find_channel(name)
+            # create format string
+            df = chan.data_format
+            fmt = (">" if df.is_be is True else "<") + (
+                "i" if df.is_signed is True else "u"
+            )
+            fmt += str(df.length // 8)
+            ba = chan.read(self.__rxbuf)  # Do local type conversion
+            data_channel_interleaved.append(np.frombuffer(ba, dtype=fmt))
+
+        return data_channel_interleaved
```

Comment:
So as I understand it, the default option is channel interleaved. I kinda missed this point, where I thought that sample interleaved is the default.

If the user wants sample interleaved they might as well do it in their user programs. I think this makes more sense!